common/math/Random.cpp

```diff
@@ -44,7 +44,7 @@ std::default_random_engine generator_;
 void InitRandom() {
   Clock clock;
   TimeStamp now;
-  clock.CurrentTime(&now);
+  clock.CurrentMonotonicTime(&now);
```

Comment:
> Don't forget to add an entry to NEWS and credit yourself in AUTHORS if you'd like.

Sounds good, will do.

> I'll make sure I either merge this or tag it as hacktoberfest-accepted before the deadline so that's one of your four sorted, hopefully you've got a few others lined up if you want your free T-shirt, or do you want to tackle some other OLA challenges?

Thank you very much, I appreciate that.  I actually could use two more PRs after this one, so if you have anything in mind that you think I could accomplish by the end of the month, I'd be happy to take a look at those too.


> I assume Monotonic time does have microsecond precision too?

At least on my Linux system, CLOCK_MONOTONIC has nanosecond precision according to clock_getres.

Reply:
> > I'll make sure I either merge this or tag it as hacktoberfest-accepted before the deadline so that's one of your four sorted, hopefully you've got a few others lined up if you want your free T-shirt, or do you want to tackle some other OLA challenges?
> 
> Thank you very much, I appreciate that.

No worries, you deserve it, I hope you've enjoyed your open source journey and it would be great if you fancy hanging around and doing the odd future pull request to scratch that itch!

> I actually could use two more PRs after this one, so if you have anything in mind that you think I could accomplish by the end of the month, I'd be happy to take a look at those too.

Here are a selection in my guess at a rough order of complexity:

Fix a compiler warning:
https://github.com/OpenLightingProject/ola/issues/1564

Port some config behaviour from one plugin to another:
https://github.com/OpenLightingProject/ola/issues/508

Or a cleanup task, definitely to target master:
https://github.com/OpenLightingProject/ola/issues/1318

Adding some extra config to the web server to host more files:
https://github.com/OpenLightingProject/ola/issues/1547

How's you're Makefile knowledge? This should be pretty trivial basically just add a phony target to run flake8, then some wrappers around it and in configure to check flake8 is available:
https://github.com/OpenLightingProject/ola/issues/1619

Perhaps part of this:
https://github.com/OpenLightingProject/ola/issues/1529

Or this:
https://github.com/OpenLightingProject/ola/issues/1480

> > I assume Monotonic time does have microsecond precision too?
> 
> At least on my Linux system, CLOCK_MONOTONIC has nanosecond precision according to clock_getres.

Excellent. I guess the quick test would be confirming that now.MicroSeconds() prints a different value each run.

Reply:
I see you've switched Random back, did it not work as intended? Can we have a comment as such if so.

Reply:
I think the change may be inconsequential in this case because random is being seeded with usecs, which repeat every second regardless of clock type.  However, if the seeding was changed in the future and secs were used, then I think the realtime secs _may_ be preferred since they do not repeat, unlike monotonic seconds which usually reset to 0 on system boot.

Reply:
Ah I didn't realise monotonic ones were that simple like a system epoch, I assumed they were still based off realtime somehow. As you can probably tell I only skim read the document.

Reply:
I couldn't tell, but it's probably worth taking a second, closer, look over all the instances we replaced with monotonic clock to make sure there are not any more gotchas like PeriodicThread.  I'll have another look at those tomorrow.

Reply:
Okay great thanks.